Parsers/ASimDhcpEvent/Parsers/vimDhcpEventInfobloxBloxOne.yaml

```diff
@@ -0,0 +1,173 @@
+Parser:
+  Title: Dhcp ASIM parser for Infoblox BloxOne
+  Version: '0.1.0'
+  LastUpdated: Jun 21 2024
+Product:
+  Name: Infoblox BloxOne
+Normalization:
+  Schema: Dhcp
+  Version: '0.1'
+References:
+- Title: ASIM Dhcp Schema
+  Link: https://aka.ms/ASimDhcpDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: Infoblox BloxOne Documentation
+  Link: https://docs.infoblox.com/space/BloxOneThreatDefense
+Description: |
+  This ASIM parser supports normalizing Dhcp logs from Infoblox BloxOne to the ASIM Dhcp normalized schema. These events are captured through the Azure Monitor Agent (AMA) which are sent by the Data Connector Service of Infoblox BloxOne.
+ParserName: vimDhcpEventInfobloxBloxOne
+EquivalentBuiltInParser: _Im_DhcpEvent_InfobloxBloxOne
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: srcipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: srchostname_has_any
+    Type: dynamic 
+    Default: dynamic([])
+  - Name: srcusername_has_any
+    Type: dynamic 
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let EventSeverityLookup = datatable(LogSeverity:string, EventSeverity:string)
+      [
+      "0", "Low",
+      "1", "Low",
+      "2", "Low",
+      "3", "Low",
+      "4", "Medium",
+      "5", "Medium",
+      "6", "Medium",
+      "7", "High",
+      "8", "High",
+      "9", "High",
+      "10", "High"
+  ];
+  let parser = (
+        starttime:datetime=datetime(null), 
+        endtime:datetime=datetime(null),
+        srcipaddr_has_any_prefix:dynamic=dynamic([]),
+        srchostname_has_any:dynamic=dynamic([]),
+        srcusername_has_any:dynamic=dynamic([]),
+        eventresult:string='*',
+        disabled:bool=false
+    ) {
+      CommonSecurityLog
+      | where not(disabled)
+          and (isnull(starttime) or TimeGenerated >= starttime) 
+          and (isnull(endtime) or TimeGenerated <= endtime)
+          and DeviceVendor == "Infoblox"
+          and DeviceEventClassID has "DHCP"
+          and ApplicationProtocol == "DHCP"
+          and (array_length(srcipaddr_has_any_prefix) == 0 or has_any_ipv4_prefix(SourceIP, srcipaddr_has_any_prefix))
+          and (array_length(srchostname_has_any) == 0 or (SourceHostName  has_any (srchostname_has_any)))
+          and array_length(srcusername_has_any) == 0
+          and ((eventresult == "*") or (eventresult == "Success"))
```

Comment:
![image](https://github.com/user-attachments/assets/48b38647-9e79-46f9-904f-237a441897d1)

Reply:
please include SrcIpAddr values in sample logs

Reply:
The SourceIP data is already present in the sample logs. There was an issue with the file name of the sample logs, but it has been resolved, and the test has now passed